x-pack/legacy/plugins/lens/public/indexpattern_plugin/dimension_panel/bucket_nesting_editor.tsx

```diff
@@ -32,35 +35,75 @@ export function BucketNestingEditor({
   const columns = Object.entries(layer.columns);
   const aggColumns = columns
     .filter(([id, c]) => id !== columnId && c.isBucketed)
-    .map(([value, c]) => ({ value, text: c.label }));
+    .map(([value, c]) => ({
+      value,
+      text: c.label,
+      fieldName: hasField(c) ? c.sourceField : '',
+    }));
 
   if (!column || !column.isBucketed || !aggColumns.length) {
     return null;
   }
 
+  const fieldName = hasField(column) ? column.sourceField : '';
+
   const prevColumn = layer.columnOrder[layer.columnOrder.indexOf(columnId) - 1];
 
   if (aggColumns.length === 1) {
     const [target] = aggColumns;
 
+    function toggleNesting() {
+      if (prevColumn) {
+        setColumns(nestColumn(layer.columnOrder, columnId, target.value));
+      } else {
+        setColumns(nestColumn(layer.columnOrder, target.value, columnId));
+      }
+    }
+
     return (
       <>
         <EuiHorizontalRule margin="m" />
-        <EuiSwitch
-          data-test-subj="indexPattern-nesting-switch"
-          label={i18n.translate('xpack.lens.xyChart.nestUnderTarget', {
-            defaultMessage: 'Nest under {target}',
-            values: { target: target.text },
+        <EuiFormRow
+          label={i18n.translate('xpack.lens.indexPattern.groupingControlLabel', {
+            defaultMessage: 'Grouping',
           })}
-          checked={!!prevColumn}
-          onChange={() => {
-            if (prevColumn) {
-              setColumns(nestColumn(layer.columnOrder, columnId, target.value));
-            } else {
-              setColumns(nestColumn(layer.columnOrder, target.value, columnId));
-            }
-          }}
-        />
+        >
+          <>
+            <EuiRadio
+              id={generator()}
+              data-test-subj="indexPattern-nesting-topLevel"
+              label={
+                column.operationType === 'terms'
+                  ? i18n.translate('xpack.lens.indexPattern.groupingOverallTerms', {
+                      defaultMessage: 'Overall top {field}',
+                      values: { field: fieldName },
+                    })
+                  : i18n.translate('xpack.lens.indexPattern.groupingOverallDateHistogram', {
+                      defaultMessage: 'Dates overall',
+                    })
+              }
+              checked={!prevColumn}
+              onChange={toggleNesting}
+            />
+            <EuiRadio
+              id={generator()}
+              data-test-subj="indexPattern-nesting-bottomLevel"
+              label={
+                column.operationType === 'terms'
+                  ? i18n.translate('xpack.lens.indexPattern.groupingSecondTerms', {
+                      defaultMessage: 'Top values for each {target}',
+                      values: { target: target.fieldName },
+                    })
+                  : i18n.translate('xpack.lens.indexPattern.groupingSecondDateHistogram', {
+                      defaultMessage: 'Dates for each {target}',
```

Comment:
I find this verbiage pretty confusing. I think `{field} for each {target}` or something would be clearer.

Reply:
I don't think it improves the readability to use field names like that- especially when the editor is showing the current field in multiple places.